feat: add an option for completing multiple words

davvid opened the pull request:
Extend the autocomplete widget so that it continues providing completions when multiple words are entered into the text field. This feature is disabled by default.

Closes: #37

JakeHandsome commented:
I tested this out and seems to work, can you update the demo app in [`demo`](https://github.com/JakeHandsome/egui_autocomplete/tree/master/demo) to show case this feature?

davvid commented:
Thanks for reviewing. I've rebased against the latest version and updated the demo app.

Just a heads-up -- the demo app hangs on me in Gnome (Wayland) on Debian/testing, but it does so ever since 10.0.0. For some reason the egui 31 update makes it freeze here. The GUI appears and is unresponsive. It's unrelated to these changes.

I'm also using egui 31 in garden-gui https://github.com/garden-rs/garden and that's working just fine, so I'm not sure why it's not working for this simple demo app. Anyways, that's a distraction so I'll ignore that for now since my suspicion is that if it's anything, it's probably related to egui itself. cheers!

JakeHandsome commented:
> Just a heads-up -- the demo app hangs on me in Gnome (Wayland) on Debian/testing, but it does so ever since 10.0.0. For some reason the egui 31 update makes it freeze here. The GUI appears and is unresponsive. It's unrelated to these changes.

Strange, I had no issues in plasma6 wayland. Seems to work as described in the demo app so I will merge this and publish a new version to crates.io. Thanks for your contribution
